events/iot.go

```diff
@@ -0,0 +1,22 @@
+package events
+
+// IoTCustomAuthorizerRequest contains data coming in to a custom IoT device gateway authorizer function.
+type IoTCustomAuthorizerRequest struct {
+	AuthorizationToken string `json:"token"`
+}
+
+// IoTCustomAuthorizerResponse represents the expected format of an IoT device gateway authorization response.
+type IoTCustomAuthorizerResponse struct {
+	IsAuthenticated          bool                   `json:"isAuthenticated"`
+	PrincipalID              string                 `json:"principalId"`
+	DisconnectAfterInSeconds int32                  `json:"disconnectAfterInSeconds"`
+	RefreshAfterInSeconds    int32                  `json:"refreshAfterInSeconds"`
+	PolicyDocuments          []string               `json:"policyDocuments"`
+	Context                  map[string]interface{} `json:"context,omitempty"`
+}
+
+// IoTCustomAuthorizerPolicy represents an IAM policy. PolicyDocuments is an array of IoTCustomAuthorizerPolicy JSON strings
```

Comment:
minor, It may make sense to add or use an existing IAMPolicy type instead of redefining the envelope. 

A README would also help to show expected usage.